provider.tf

```diff
@@ -1,4 +1,25 @@
- 
+
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = ">=3.0, ~> 4.20"
```

Comment:
Shouldn't this be `version = "~> 4.20"`?

Reply:
so far we know the module works fine with aws provider 3.x and 4.x, so we want to keep the provider version between 3.x and 4.x, to avoid problems